controllers/__init__.py

```diff
@@ -273,38 +273,28 @@ def bulk_delete_link_metadata_key(
         return self.db.links.update_many({"_id": {"$in": link_ids}},
                                          update_expr)
 
-    def bulk_upsert_interface_details(
-        self, ids_details: List[Tuple[str, dict]]
+    def upsert_interface_details(
+        self,
+        id_: str,
+        available_tags: dict[list[list[int]]],
+        tag_ranges: dict[list[list[int]]],
     ) -> Optional[dict]:
         """Update or insert interfaces details."""
         utc_now = datetime.utcnow()
-        ops = []
-        for _id, detail_dict in ids_details:
-            ops.append(
-                UpdateOne(
-                    {"_id": _id},
-                    {
-                        "$set": InterfaceDetailDoc(
-                            **{
-                                **detail_dict,
-                                **{
-                                    "updated_at": utc_now,
-                                    "_id": _id,
-                                },
-                            }
-                        ).dict(exclude={"inserted_at"}),
-                        "$setOnInsert": {"inserted_at": utc_now},
-                    },
-                    upsert=True,
-                ),
-            )
-
-        with self.interface_details_lock:
```

Comment:
`self.interface_details_lock` this attribute initialization in the constructor can be removed too

Reply:
Deleted but keeping an eye on it in case it needs to comeback.